core/src/impl/Kokkos_ViewMapping.hpp

```diff
@@ -2978,19 +2980,22 @@ struct ViewValueFunctor<DeviceType, ValueType, false /* is_scalar */> {
   std::enable_if_t<!(std::is_trivial<Dummy>::value &&
                      std::is_trivially_copy_assignable<ValueType>::value)>
   construct_dispatch() {
-    parallel_for_implementation(false);
+    parallel_for_implementation<ConstructTag>();
   }
 
-  void parallel_for_implementation(bool arg) {
-    destroy = arg;
+  template <typename Tag>
+  void parallel_for_implementation() {
     if (!space.in_parallel()) {
-      PolicyType policy(0, n);
+      using PolicyType =
+          Kokkos::RangePolicy<ExecSpace, Kokkos::IndexType<int64_t>, Tag>;
+      PolicyType policy(space, 0, n);
       std::string functor_name;
       uint64_t kpID = 0;
       if (Kokkos::Profiling::profileLibraryLoaded()) {
         functor_name =
-            (destroy ? "Kokkos::View::destruction [" + functor_name + "]"
-                     : "Kokkos::View::initialization [" + functor_name + "]");
+            (std::is_same_v<Tag, DestroyTag>
+                 ? "Kokkos::View::destruction [" + functor_name + "]"
+                 : "Kokkos::View::initialization [" + functor_name + "]");
         Kokkos::Profiling::beginParallelFor(
             "Kokkos::View::initialization [" + functor_name + "]",
```

Comment:
It looks like this should just be `functor_name` now, since initialization/destruction was specified in it just above